cli/build.js

```diff
@@ -1,44 +1,43 @@
 import fs from 'fs'
 
 import { compileSchema, compileTs } from './compile'
-import { copyByExtension } from './copy'
+import {
+    copyByExtension,
+    copyByExtensionWithExtensionReplacement,
+} from './copy'
 import * as paths from './paths'
 import { existsSyncWithExtension } from './utils'
 
 if (fs.existsSync(paths.app.schema)) {
-    compileSchema(paths.app.schema, paths.mould.components, ([s, ns]) =>
-        console.log(
-            `Compiled Mould Components successfully in ${s}s ${ns / 1e6}ms`
-        )
-    )
+    const time = process.hrtime()
+
+    Promise.all([
+        compileSchema(paths.app.schema, paths.mould.components),
+        existsSyncWithExtension(paths.app.mouldDirectory, '.ts') &&
+            copyByExtension(
+                paths.app.mouldDirectory,
+                paths.mould.componentsDirectory,
+                '.ts'
+            ),
+        existsSyncWithExtension(paths.app.mouldDirectory, '.js') &&
+            copyByExtensionWithExtensionReplacement(
+                paths.app.mouldDirectory,
+                paths.mould.componentsDirectory,
+                '.js',
+                '.ts'
+            ),
```

Comment:
We copy `.js` files into the `/.components` directory with replacing their extensions to `.ts` in order to compile the entire `/.components` directory using `tsc`, for instance

### A TypeScript project 
```
/node_modules
  /@mouldjs
    /mould
      /.components
        index.js
/mould
  .mould
  resolvers.ts
package.json
tsconfig.json
``` 
The `mould build` command behaves as follows
1. Prepares the `/.components` directory for the `tsc`
1.1. Compiles Mould Schema (`.mould`) into React Components (`/.components/index.tsx`)
1.2. **Copies `*.ts` files into the `/.components` directory**
2. Runs `tsc`

As a result, `@mouldjs/mould` has all the compiled components and resolvers along with types
```
/node_modules
  /@mouldjs
    /mould
      /.components
        index.d.ts
        index.js
        index.tsx
        resolvers.d.ts
        resolvers.js
        resolvers.ts
/mould
  .mould
  resolvers.ts
package.json
tsconfig.json
``` 

### A JavaScript project 
```
/node_modules
  /@mouldjs
    /mould
      /.components
        index.js
/mould
  .mould
  resolvers.js
package.json
``` 
The `mould build` command behaves as follows
1. Prepares the `/.components` directory for the `tsc`
1.1. Compiles Mould Schema (`.mould`) into React Components (`/.components/index.tsx`)
1.2. **Copies `*.js` files into the `/.components` directory with replacing their type with `.ts`**
2. Runs `tsc`

As a result, `@mouldjs/mould` has all the compiled components and resolvers
```
/node_modules
  /@mouldjs
    /mould
      /.components
        index.d.ts
        index.js
        index.tsx
        resolvers.d.ts
        resolvers.js
        resolvers.ts
/mould
  .mould
  resolvers.ts
package.json
tsconfig.json
```